Modularize content management guide

maximiliankolb opened the pull request:
This PR modularizes the content management guide. Luckily it's much small than the other three PRs.

One thing I've noted: there's three files to create activation keys by now and two to create a content view. Maybe we should do something about this in the future. However, **this** PR is merely modularizing content.

Cherry-pick into:

* [x] Foreman 3.1

maximiliankolb commented:
Rebased to master.

Lennonka commented on guides/common/assembly_basic-content-management-workflow.adoc:
```diff
@@ -0,0 +1,29 @@
+include::modules/con_basic-content-management-workflow.adoc[]
```
Perhaps this assembly should be included in the `master.adoc`?

Lennonka replied:
@maximiliankolb And this one

maximiliankolb replied:
Fixed. I had (by mistake) included the concept instead of the assembly.

Lennonka commented on guides/common/assembly_converting-a-host-with-convert2rhel.adoc:
```diff
@@ -1,56 +1,10 @@
 ifdef::context[:parent-context: {context}]
 
-[id="assembly_converting-a_host-with-convert2rhel_{context}"]
-= Converting a Host to {RHEL} with Convert2RHEL
-
-Convert2RHEL enables the conversion of {RHEL} derivative distributions into a supportable {RHEL} state while retaining existing applications and configurations.
-
-The conversion process is similar to a minor release upgrade of {RHEL} in which every RPM package on the system is replaced.
-Third-party packages and non-{Team} packages not available in {RHEL} are not replaced.
-
-Convert2RHEL removes unnecessary packages such as logos, or packages known to cause issues during the conversion, and replaces the `CentOS-release` package with the `rhel-release` package.
-It subscribes the system to {ProjectName} and replaces all packages signed by CentOS.
-ifdef::satellite[]
-For {Project}, it also subscribes the system to {Team} Subscription Management.
-endif::[]
-
-The amount of time taken to do this can vary depending on how many packages need to be replaced, network speed, storage speed, and similar variables.
-
-Use the global registration template to register and subscribe your system before the conversion.
-For more information, see {ManagingHostsDocURL}registering-a-host_managing-hosts[Registering a Host to {ProjectName} Using the Global Registration Template] in the _Managing Hosts_ guide.
-
-To convert hosts to {RHEL} with convert2RHEL, you must complete the following procedures:
-
-. xref:installing-convert2rhel_{context}[].
-. xref:Creating_an_Activation_Key_{context}[].
-. xref:preparing-the-host-group_{context}[].
-. xref:registering-hosts-for-conversion_{context}[].
-. xref:converting-hosts-with-convert2rhel_{context}[]
-
```
I'd keep the title and intro in the assembly and put only the Prerequisites into a separate concept module.

Lennonka commented on guides/common/assembly_importing-content.adoc:
```diff
@@ -1,18 +1,4 @@
-
-* xref:Managing_ISO_Images_{context}[]
-* xref:Managing_Custom_File_Type_Content_{context}[]
+include::modules/con_importing-content.adoc[]
```
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_importing-kickstart-repositories.adoc:
```diff
@@ -1,18 +1,6 @@
-ifndef::satellite[]
-You can follow these procedures for {RHEL} derivatives such as Alma Linux, CentOS, Oracle Linux, and Rocky Linux, too.
-endif::[]
+include::modules/con_importing-kickstart-repositories.adoc[]
```
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-activation-keys.adoc:
```diff
@@ -1,47 +1,4 @@
-[id="Managing_Activation_Keys_{context}"]
-= Managing Activation Keys
```
The title and perhaps an intro paragraph should stay in the assembly.
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-application-life-cycles.adoc:
```diff
@@ -1,7 +1,4 @@
-= Managing Application Life Cycles
-
-This chapter outlines the application life cycle in {Project} and how to create and remove application life cycles for {Project} and {SmartProxy}.
+include::modules/con_managing-application-life-cycles.adoc[]
```
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-container-images.adoc:
```diff
@@ -1,11 +1,4 @@
-ifndef::orcharhino[]
-For information about containers, see https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/getting_started_with_containers/index[Getting Started with Containers] in _Red Hat Enterprise Linux Atomic Host 7_.
-endif::[]
+include::modules/con_managing-container-images.adoc[]
```
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-content-views.adoc:
```diff
@@ -1,52 +1,4 @@
-[id="Managing_Content_Views_{context}"]
-= Managing Content Views
-
-{ProjectNameX} uses Content Views to create customized repositories from the repositories.
-To do this, you must define which repositories to use and then apply certain filters to the content.
-These filters include both package filters, package group filters, errata filters, and module stream filters.
-You can use Content Views to define which software versions a particular environment uses.
-For example, a _Production_ environment might use a Content View containing older package versions, while a _Development_ environment might use a Content View containing newer package versions.
```
Title and an introduction should stay in the assembly.
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-custom-file-type-content.adoc:
```diff
@@ -1,17 +1,4 @@
-You can create a file type repository in {ProjectServer} only in a {customproduct}, but there is flexibility in how you create the file type repository.
-You can create an independent file type repository in a directory on the system where {Project} is installed, or on a remote HTTP server, and then synchronize the contents of that directory into {Project}.
-This method is useful when you have multiple files to add to a {Project} repository.
+include::modules/con_managing-custom-file-type-content.adoc[]
```
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-errata.adoc:
```diff
@@ -1,55 +1,4 @@
-[id="Managing_Errata_{context}"]
-= Managing Errata
-
-ifdef::satellite[]
-As a part of Red Hat's quality control and release process, we provide customers with updates for each release of official Red Hat RPMs.
-Red Hat compiles groups of related package into an *erratum* along with an advisory that provides a description of the update.
-endif::[]
-ifndef::satellite[]
-{RHEL} users receive errata for each release of official Red Hat RPMs.
-SUSE Linux Enterprise Server users receive errata for each release of official SUSE RPMs.
-endif::[]
-ifdef::orcharhino[]
-Additionally, ATIX provides Debian and Ubuntu users with errata.
-endif::[]
```
Perhaps this should stay in the assembly.
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-isos-and-files.adoc:
```diff
@@ -1,8 +1,4 @@
-
-You can use {Project} to store ISO images, either from Red{nbsp}Hat's Content Delivery Network or other sources.
-You can also upload other files, such as virtual machine images, and publish them in repositories.
+include::modules/con_managing-iso-images.adoc[]
```
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-locations.adoc:
```diff
@@ -1,11 +1,4 @@
-
-* Locations are based on physical or geographical settings.
-* Locations have a hierarchical structure.
+include::modules/con_managing-locations.adoc[]
```
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_managing-organizations.adoc:
```diff
@@ -1,41 +1,4 @@
-[id="Managing_Organizations_{context}"]
-= Managing Organizations
-
-Organizations divide {ProjectName} resources into logical groups based on ownership, purpose, content, security level, or other divisions.
-ifdef::satellite[]
-You can create and manage multiple organizations through {ProjectName}, then divide and assign your Red Hat subscriptions to each individual organization.
-endif::[]
-ifndef::satellite[]
-You can create and manage multiple organizations through {ProjectName}, then divide and assign subscriptions to each individual organization.
-endif::[]
-This provides a method of managing the content of several individual organizations under one management system.
```
Perhaps this should stay.
Mark -- see my comment for `assembly_synchronizing-content-between-foreman.adoc`

Lennonka commented on guides/common/assembly_synchronizing-content-between-foreman.adoc:
```diff
@@ -1,14 +1,10 @@
-= Synchronizing Content Between {ProjectServer}s
-
-{ProjectName} uses Inter-Server Synchronization (ISS) to synchronize content between two {ProjectServer}s including those that are airgapped.
+include::modules/con_synchronizing-content-between-servers.adoc[]
```
I have reviewed all assemblies once more and I suggest we reconsider this approach in general.

Lennonka reviewed (changes requested):
I don't understand why the headings and introductions of assemblies should be put into modules. Can we discuss this approach, please?

maximiliankolb commented:
We basically need this in downstream, but I have good reasons to change this upstream:

I strongly believe we should not have headings/text/any content in general in assemblies, and here's why:

1. Once we move content to separate files, we can run certain bulk commands on all files more easily, based on the assumption that "real" content is only in `guides/common/modules/` and split into different types of content (`con`, `proc`, `ref`, etc).

2. I see two future benefits content wise: First, we can later on add meta data based on the type of content (coming from the file name prefix), and we can create more diverse guides and assemblies for more than one operating system/configuration management solution/compute resource. I think the community will be happy to receive more specific guides depending on their OS.

3. Personally, I can see more clearly what's included, because we have a perfect 1:1 mapping from headings on docs.theforeman.org to indiviual files. This also makes sense from the perspektive of the necessary level offsets. I hope this will also encourage community/developer contributions, because existing content is even easier to find purely based on the title.

4. I also think that there will be more content in the future to avoid having two headings without content in between.

5. Structurally speaking, this means we split content and structure/ToC/order of items; which in some case in necessary for upstream/downstream. Until now, it was mostly based on Red Hat's original documentation contribution, and sometimes missing procedures and concepts on other operating systems such as Ubuntu or plugins such as Salt.

FYI: We have merged something like this in the past ([example](https://github.com/theforeman/foreman-documentation/blame/0e39b8c54d2d61e2207c44f2ae92609181afb2b2/guides/common/modules/con_managing-compliance-policies.adoc)).

Lennonka commented:
Oki, I see you have thought this through, so I'm going to trust it's a way to go and I'll see what I can learn from it. Let me re-do the review with this knowledge.

Lennonka commented on guides/common/assembly_converting-a-host-with-convert2rhel.adoc:
```diff
@@ -1,56 +1,10 @@
-== Creating an Activation Key for the Hosts
-
-See xref:Creating_an_Activation_Key_{context}[].
+include::modules/proc_creating-an-activation-key-short-2.adoc[leveloffset=+1]
```
@maximiliankolb Can you please check this one?

maximiliankolb replied:
Fixed!

Lennonka commented:
@maximiliankolb So, please ignore my comments concerning the assembly headings/intros, and just check the two other comments I tagged you in. Otherwise all good.

maximiliankolb commented:
> @maximiliankolb So, please ignore my comments concerning the assembly headings/intros, and just check the two other comments I tagged you in. Otherwise all good.

I will resolve both issues in the morning!

maximiliankolb commented:
Rebased to master & resolved all issues @Lennonka

Lennonka commented:
@maximiliankolb Once we unify the emphasis of the guide references, we're good to merge.

maximiliankolb commented on guides/doc-Deploying_on_AWS/topics/AWS_Use_Case_Considerations.adoc:
```diff
@@ -51,7 +51,7 @@ ifndef::foreman-deb[]
 When you set the download policy to *On Demand*, content syncs to {ProjectServer} or {SmartProxyServer} when a content host requests it.
 
-For more information, see {ContentManagementDocURL}Importing_Content[Importing Content] in the _Content Management Guide_.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in the _Content Management_ guide.
```
Applied.

maximiliankolb commented:
> @maximiliankolb Once we unify the emphasis of the guide references, we're good to merge.

I've created an issue for this: https://github.com/theforeman/foreman-documentation/issues/954

I think we're good to go.
